module-4/backend/week_four.md

```diff
@@ -5,18 +5,43 @@ Fork this respository. Answer the questions to the best of your ability. Try to
+
 * What are the stages of a card when using Pivotal Tracker?
+Start, finish, deliver, complete.
+
 * What are the four types of cards available in Pivotal Tracker?
+Feature, bug, chore, release.
 * What's your favorite project management tool?
+So far, Waffle, for it's simplicity and connection to Github.
+
 * Why do we create staging environments?
+In order to test how our code will work in production on different machines / browsers, without having to push to production. This also enables testing and approval by product owners and potentially the client.
+
 * What are the characteristics of a good README (in your opinion)?
+It explains both the technical set up and the qualititative purpose of the project; in open source, how to contribute.
+
 * What's one main improvement you're going to make to your code regarding accessibility issues?
+I want to enable tab completion and navigation in all my websites. I also think that in terms of presenting content to the user, it's important both for that content to be accessible to those with physical or mental challenges, but also because optimizing for accessibility also inherently increases usability by non-challenged people. 
+
 * What are some basic security concerns to be aware of when building applications?
+SQL injection, man in the middle, click jacking.
+
 * Why is continuous integration helpful/important?
+It is very helpful for ensuring that all the tests are run properly prior to merging into master or pushing to production. This enables a team to be confident that their testing suite is constantly being run and that any errors will be caught before pushing to production. Especially as a project grows, this also allows the full testing suite to be run on a virtual machine, rather than locally, saving time.  
+
 * What are some continuous integration tools?
+TravisCI
 
 #### Review  
 
 * What are some characteristics of "good" git workflow?
+Pulling master into the branch to be merged prior to merging, writing good git commit messages, tagging people in pull requests and requiring two people to approve the PR prior to merging. Leaving comments on code and running a branches test suite locally prior to 
+
 * What are the four fundamental concepts of object oriented programming?
+Encapsulation, abstraction, inheritance, and polymorphism. 
+
 * What's a module in Ruby and what's the difference between a class and a module?
+A module is a set of methods that are accessible anywhere they are explicitly inherited, but it does not allow the instantiation of an object or calling a overarching method on a model of some behavior. Classes, on the other hand, are used both to allow a set of methods that are inheritable wherever they are explicitly inherited, but it's primary function is to model a set of functionality and enable the instantiation of objects which can be accessed and modified according to methods defined in the class. 
```

Comment:
modules aren't inherited by classes - they are "mixed in" - we include them within the specific classes we want to have access to them within 👍